haskell-src/Concordium/Logger.hs

```diff
@@ -102,6 +103,21 @@ runSilentLogger = flip runLoggerT (\_ _ _ -> pure ())
 
 ------------------------------------------------------------------------------
 
+-- * The @Loggable@ class
+
+-- | Class for structured log messages.
+class Loggable event where
```

Comment:
You can grep for the logevent - and then grep on that event. Admittedly it is one more grep that is necessary. But this way of logging shouldn't really put us in a worse situation as to the one you described above. 

By having this `Loggable` class and the associated events we can easier change the messages in the future and make sure that they adhere to some standard. 

Also with this approach we can also easier e.g. use `Text` instead of `String`s - which should be beneficial at least for trace logging.

Reply:
No, I looked at how this is used as well, and this is just more error prone. You are defining variants for each different log message.

This is not a good solution.